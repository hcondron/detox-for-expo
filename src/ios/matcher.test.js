describe('ios matcher', async () => {
  let m;

  beforeEach(() => {
    m = require('./matchers');
  });

  // These snapshots preserve the output of the matchers against unwanted change
  it("snapshots", () => {
    const textBasedMatchers = ['LabelMatcher', 'IdMatcher', 'TypeMatcher', 'TextMatcher', 'ValueMatcher'];
    const chainableMatchers = ['withAncestor', 'withDescendant', 'and', 'not'];
    const noArgsMatchers = ['VisibleMatcher', 'NotVisibleMatcher', 'ExistsMatcher', 'NotExistsMatcher'];

    textBasedMatchers.forEach(matcher => {
      expect(new m[matcher]("mySelector")._call()).toMatchSnapshot();
    });

    chainableMatchers.forEach(matcher => {
      expect(new m.LabelMatcher("mySelector")[matcher](new m.LabelMatcher("myOtherSelector"))._call()).toMatchSnapshot()
    });

    noArgsMatchers.forEach(matcher => {
      expect(new m[matcher]()._call()).toMatchSnapshot();
    });
  });
});