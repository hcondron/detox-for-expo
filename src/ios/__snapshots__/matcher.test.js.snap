// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ios matcher snapshots 1`] = `
Object {
  "args": Array [
    "mySelector",
  ],
  "method": "detox_matcherForAccessibilityLabel:",
  "target": Object {
    "type": "Class",
    "value": "GREYMatchers",
  },
}
`;

exports[`ios matcher snapshots 2`] = `
Object {
  "args": Array [
    "mySelector",
  ],
  "method": "matcherForAccessibilityID:",
  "target": Object {
    "type": "Class",
    "value": "GREYMatchers",
  },
}
`;

exports[`ios matcher snapshots 3`] = `
Object {
  "args": Array [
    "mySelector",
  ],
  "method": "detoxMatcherForClass:",
  "target": Object {
    "type": "Class",
    "value": "GREYMatchers",
  },
}
`;

exports[`ios matcher snapshots 4`] = `
Object {
  "args": Array [
    "mySelector",
  ],
  "method": "detoxMatcherForText:",
  "target": Object {
    "type": "Class",
    "value": "GREYMatchers",
  },
}
`;

exports[`ios matcher snapshots 5`] = `
Object {
  "args": Array [
    "mySelector",
  ],
  "method": "matcherForAccessibilityValue:",
  "target": Object {
    "type": "Class",
    "value": "GREYMatchers",
  },
}
`;

exports[`ios matcher snapshots 6`] = `
Object {
  "args": Array [
    Object {
      "type": "Invocation",
      "value": Object {
        "args": Array [
          "mySelector",
        ],
        "method": "detox_matcherForAccessibilityLabel:",
        "target": Object {
          "type": "Class",
          "value": "GREYMatchers",
        },
      },
    },
    Object {
      "type": "Invocation",
      "value": Object {
        "args": Array [
          "myOtherSelector",
        ],
        "method": "detox_matcherForAccessibilityLabel:",
        "target": Object {
          "type": "Class",
          "value": "GREYMatchers",
        },
      },
    },
  ],
  "method": "detoxMatcherForBoth:andAncestorMatcher:",
  "target": Object {
    "type": "Class",
    "value": "GREYMatchers",
  },
}
`;

exports[`ios matcher snapshots 7`] = `
Object {
  "args": Array [
    Object {
      "type": "Invocation",
      "value": Object {
        "args": Array [
          "mySelector",
        ],
        "method": "detox_matcherForAccessibilityLabel:",
        "target": Object {
          "type": "Class",
          "value": "GREYMatchers",
        },
      },
    },
    Object {
      "type": "Invocation",
      "value": Object {
        "args": Array [
          "myOtherSelector",
        ],
        "method": "detox_matcherForAccessibilityLabel:",
        "target": Object {
          "type": "Class",
          "value": "GREYMatchers",
        },
      },
    },
  ],
  "method": "detoxMatcherForBoth:andDescendantMatcher:",
  "target": Object {
    "type": "Class",
    "value": "GREYMatchers",
  },
}
`;

exports[`ios matcher snapshots 8`] = `
Object {
  "args": Array [
    Object {
      "type": "Invocation",
      "value": Object {
        "args": Array [
          "mySelector",
        ],
        "method": "detox_matcherForAccessibilityLabel:",
        "target": Object {
          "type": "Class",
          "value": "GREYMatchers",
        },
      },
    },
    Object {
      "type": "Invocation",
      "value": Object {
        "args": Array [
          "myOtherSelector",
        ],
        "method": "detox_matcherForAccessibilityLabel:",
        "target": Object {
          "type": "Class",
          "value": "GREYMatchers",
        },
      },
    },
  ],
  "method": "detoxMatcherForBoth:and:",
  "target": Object {
    "type": "Class",
    "value": "GREYMatchers",
  },
}
`;

exports[`ios matcher snapshots 9`] = `
Object {
  "args": Array [
    Object {
      "type": "Invocation",
      "value": Object {
        "args": Array [
          "mySelector",
        ],
        "method": "detox_matcherForAccessibilityLabel:",
        "target": Object {
          "type": "Class",
          "value": "GREYMatchers",
        },
      },
    },
  ],
  "method": "detoxMatcherForNot:",
  "target": Object {
    "type": "Class",
    "value": "GREYMatchers",
  },
}
`;

exports[`ios matcher snapshots 10`] = `
Object {
  "args": Array [],
  "method": "matcherForSufficientlyVisible",
  "target": Object {
    "type": "Class",
    "value": "GREYMatchers",
  },
}
`;

exports[`ios matcher snapshots 11`] = `
Object {
  "args": Array [],
  "method": "matcherForNotVisible",
  "target": Object {
    "type": "Class",
    "value": "GREYMatchers",
  },
}
`;

exports[`ios matcher snapshots 12`] = `
Object {
  "args": Array [],
  "method": "matcherForNotNil",
  "target": Object {
    "type": "Class",
    "value": "GREYMatchers",
  },
}
`;

exports[`ios matcher snapshots 13`] = `
Object {
  "args": Array [],
  "method": "matcherForNil",
  "target": Object {
    "type": "Class",
    "value": "GREYMatchers",
  },
}
`;
